util/internal/integrationtest/mapper/orm/src/main/java/org/hibernate/search/util/impl/integrationtest/mapper/orm/DatabaseContainer.java

```diff
@@ -0,0 +1,428 @@
+			HibernateSearchJdbcDatabaseContainer container(Path dockerfile, String name) {
+				return new HibernateSearchJdbcDatabaseContainer(
+						dockerfile, name,
+						"com.microsoft.sqlserver.jdbc.SQLServerDriver",
+						"jdbc:sqlserver://%s:%d;databaseName=tempdb",
+						1433,
+						"SA",
+						"ActuallyRequired11Complexity",
+						"select 1"
+				).withEnv( "ACCEPT_EULA", "Y" )
+						.withEnv( "SA_PASSWORD", "ActuallyRequired11Complexity" )
+						.withStartupTimeout( EXTENDED_TIMEOUT );
+			}
+		},
+		COCKROACHDB {
+			@Override
+			String dialect() {
+				return org.hibernate.dialect.CockroachDialect.class.getName();
+			}
+
+			@Override
+			HibernateSearchJdbcDatabaseContainer container(Path dockerfile, String name) {
+				return new HibernateSearchJdbcDatabaseContainer(
+						dockerfile, name,
+						"org.postgresql.Driver",
+						"jdbc:postgresql://%s:%d/defaultdb?sslmode=disable",
+						26257,
+						"root",
+						"",
+						"select 1",
+						new HttpWaitStrategy()
+								.forPath( "/health" )
+								.forPort( 8080 )
+								.forStatusCode( 200 )
+				).withCommand( "start-single-node --insecure" );
+			}
+		};
+
+		Configuration configuration(JdbcDatabaseContainer<?> container) {
+			return new Configuration(
+					dialect(),
+					container.getDriverClassName(),
+					container.getJdbcUrl(),
+					container.getUsername(),
+					container.getPassword(),
+					""
+			);
+		}
+
+		abstract String dialect();
+
+		abstract HibernateSearchJdbcDatabaseContainer container(Path dockerfile, String name);
+
+		static SupportedDatabase from(String name) {
+			for ( SupportedDatabase database : values() ) {
+				if ( name.toLowerCase( Locale.ROOT ).contains( database.name().toLowerCase( Locale.ROOT ) ) ) {
+					return database;
+				}
+			}
+			throw new IllegalStateException( "Unsupported database requested: " + name );
+		}
+	}
+
+	private static class HibernateSearchJdbcDatabaseContainer
+			extends JdbcDatabaseContainer<HibernateSearchJdbcDatabaseContainer> {
+
+		private final String driverClassName;
+		private final String jdbcUrlPattern;
+		private final int port;
+		private final String username;
+		private final String password;
+		private final String testQueryString;
+		private final Optional<WaitStrategy> customWaitStrategy;
+
+		public HibernateSearchJdbcDatabaseContainer(Path dockerfile, String name, String driverClassName, String jdbcUrlPattern,
+				int port, String username, String password, String testQueryString) {
+			this( dockerfile, name, driverClassName, jdbcUrlPattern, port, username, password, testQueryString, null );
+		}
+
+		public HibernateSearchJdbcDatabaseContainer(Path dockerfile, String name, String driverClassName, String jdbcUrlPattern,
+				int port, String username, String password, String testQueryString, WaitStrategy waitStrategy) {
+			// IMPORTANT: we do not want to delete the image on exit as then we cannot use container reuse.
+			// (these two options are somewhat mutually exclusive).
```

Comment:
this is what was causing Oracle tests to fail. While at it I've added a custom wait condition; this way DB startup is cleaner for some of the containers.